src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,274 @@
+namespace Nancy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Nancy.Responses.Negotiation;
+
+    /// <summary>
+    /// Represents one of possibly many RFC 5899 HTTP Links contained in a <see cref="HttpLinkHeader"/>.
+    /// </summary>
+    public class HttpLink : Dictionary<string, object>, IEquatable<HttpLink>
+    {
+        private readonly Uri targetUri;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpLink" /> class.
+        /// </summary>
+        /// <param name="targetUri">The target URI of the link.</param>
+        /// <param name="relation">The relation that identifies the semantics of the link.</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(string targetUri, string relation)
+            : this(ParseUri(targetUri), relation, null, null)

```

Comment:
Could `type` and `title` be `string.Empty` instead? I'm generally not a fan of keeping `null` around :smile: Same goes for the other constructors. Might also be worth doing a null check before the values are used, just in case someone tries to pass `null`.